sympy/stats/stochastic_process_types.py

```diff
@@ -676,31 +771,829 @@ def _transient2absorbing(self):
+
+        .. [1] https://www.probabilitycourse.com/chapter11/11_2_6_stationary_and_limiting_distributions.php
+        .. [2] https://galton.uchicago.edu/~yibi/teaching/stat317/2014/Lectures/Lecture4_6up.pdf
+
+        See Also
+        ========
+
+        sympy.stats.stochastic_processes_types.DiscreteMarkovChain.stationary_distribution
+        """
+        trans_probs = self.transition_probabilities
+        n = self.num_states
+
+        if not self._is_numeric():
+            return self.stationary_distribution()
+
+        comm_classes = self.communication_classes()
+
+        # if irreducible and aperiodic
+        if (len(comm_classes) == 1) and (comm_classes[0].period == 1):
+            return self.stationary_distribution()
+
+        # limits do not work well with matrices
+        # I think this is still the stationary distribution
+        # if all recurrent classes are aperiodic.
+        # I have no sources to back this up though.
+        if all([c.period == 1 for c in comm_classes if c.recurrent]):
+            return self.stationary_distribution()
+
+        # If the chain is reducible or periodic, the
+        # limiting distribution might not be unique or exist.
+        # It would depend on the time 0 marginal
+        # state probability vector.
+
+        if n == 0:  # a zero by zero matrix to any power is just itself
+            return Matrix([[]])
+
+        p0_is_given = p0 is not None  # whether or not the user gave their own p0.
+        # using p0_is_given will give more refined AccumBounds in the example.
+        # If we treat p0 the same way as whether it was given or not,
+        # the lower and upper bounds will be wider than necessary for a given p0.
+        # The last example in the docstring would give
+        # Matrix([[AccumBounds(0, 1), AccumBounds(0, 1)]]).
+        # so if p0 is given, we add it into the mix as soon as possible.
+
+        if not p0_is_given:
+            p0 = MatrixSymbol('p_0', 1, n)  # implicitly row-stochastic
+
+        _n = Dummy("n", positive=True, integer=True)
+        if p0_is_given:
+            P_to_the_n = p0 * Matrix(trans_probs) ** _n
+        else:
+            P_to_the_n = Matrix(trans_probs) ** _n
+
+        for row in range(P_to_the_n.shape[0]):
+            for col in range(P_to_the_n.shape[1]):
+                try:  # try to limit pn but it sometimes does not work
+                    P_to_the_n[row, col] = limit_seq(P_to_the_n[row, col], _n)
+                except NotImplementedError:
+                    pass
+
+        if p0_is_given:
+            pn = P_to_the_n
+        else:
+            pn = p0 * P_to_the_n
+        return pn
+
+    def communication_classes(self):
+        """
+        Returns the list of communication classes that partition
+        the states of the markov chain. It will return None if
+        no transition matrix was given.
+
+        Returns
+        =======
+
+        classes : List of CommunicationClass, optional
```

Comment:
A simple set of states should work.

Reply:
The order matters for markov chains since it's used to index the transition matrix. I thought maybe a set of lists could work, but even then the order will be lost. 

For example given a 3x3 matrix M, its canonical form A, with the two communication classes 0 and 1, 2. Let's say both of these classes are recurrent. Then we have [[1, 2], [0]] and [[0], [1, 2]] both have different meanings when looking at A.

Reply:
Is it possible the idea of strongly connected components to find sets of states which are communicating?

Reply:
Yes. That's hopefully what I'll do as suggested by Oscar. Once obtained though, the states would then need to have a chosen order. It doesn't really matter what the order is until we define A.